case-lib/lib.sh

```diff
@@ -253,6 +253,50 @@ func_lib_get_tplg_path()
     return 0
 }
 
+# We must not quote SOF_ALSA_OPTS and disable SC2086 below for two reasons:
+#
+# 1. We want to support multiple parameters in a single variable, in
+#    other words we want to support this:
+#
+#      SOF_ALSA_OPTS="--foo --bar"
+#      aplay $SOF_ALSA_OPTS ...
+#
+# 2. aplay does not ignore empty arguments anyway, in other words this
+#    does not work anyway:
+#
+#      SOF_ALSA_OPTS=""
+#      aplay "$SOF_ALSA_OPTS" ...
+#
+# This is technically incorrect because it means our SOF_ALSA_OPTS
+# cannot support whitespace, for instance this would be split in two
+# options: --bar="aaaa and bbbb"
+#
+#      SOF_ALSA_OPTS='--foo --bar="aaaa bbbb"'
+#
+# To do this "correctly" SOF_ALSA_OPTS should be an array.
+# From https://mywiki.wooledge.org/BashGuide/Arrays "The only safe way
+# to represent multiple string elements in Bash is through the use of
+# arrays."
+# However 1. it would complicate the user interface 2. We don't seem to
+# need arguments with whitespace or globbing characters.
+
+# Default value, can be overriden by the environment. Empty string does
+# override.
+: "${SOF_ALSA_OPTS=--fatal-errors}"
```

Comment:
@marc-hb maybe consider to put this into the `case-lib/config.sh` file? That config file is designed for overridable ENV values.

Reply:
I like config files but I see a number of issues with the `case-lib/config.sh` design. I'll remove the default value here and resubmit so we can postpone the potentially complex discussion about config files and focus on these wrappers first. One change  at a time!

Some issues with `case-lib/config.sh`:
- Except for two actual values, all the rest is made of no-op `AAAA=${AAAA:-}` statements. That's very misleading from a coding perspective.
- What would be more useful is a local config file that is not in git so it doesn't pollute `git diff` and `git status` (and even [`git add *`](https://github.com/thesofproject/rimage/commit/0d2af68a9e2bc) - never do that)
- I'm not even sure a single file collecting all default values is a good idea. Most software seems to have good defaults even when run with empty config files.

Reply:
I create a new issue https://github.com/thesofproject/sof-test/issues/502 for the later discussion. Nice point here